src/plugins/intel_gpu/src/graph/impls/ocl/concatenation.cpp

```diff
@@ -72,11 +72,38 @@ struct concatenation_impl : typed_primitive_impl_ocl<concatenation> {
 
         return {params, optional_params};
     }
+
+    void update_dispatch_data(const kernel_impl_params& impl_param) override {
+        auto kernel_params = get_kernel_params(impl_param);
```

Comment:
auto kernel_params = get_kernel_params(impl_param, true);

Reply:
Looks like that's not enough to enable empty tensor support for shape agnostic multi-stage kernels. 

For example for 2 stages in concat kernel with empty first input (case from unit tests), `get_kernels_source()` returned 2 kernels as the impl shape agnostic and `skip_execution` was equal to `false` for both kernels. Then `update_kernels_list_to_skip` marked first kernel to be skipped due to gws=0, thus `set_arguments()` and `execute()` tried to skip it, but since index of the kernel taken from `_kernels` was independently incremented after actual kernel execution, second stage actually ran first kernel. 

So I removed `skip_execution` check in `get_kernels_source()` and made `set_arguments()/execute()` method use same idx for `_kernels` and `_kernel_data.kernels`

Reply:
Oh I got it, having difference indexing for kernel and kernel data is problematic for "multi staged shape agnostic kernels.." I see..